src/goto-instrument/cover.cpp

```diff
@@ -68,6 +105,112 @@ class basic_blockst
   }
 };
 
+bool coverage_goalst::get_coverage_goals(
+  const std::string &coverage_file,
+  message_handlert &message_handler,
+  coverage_goalst &goals)
+{
+  jsont json;
+  source_locationt source_location;
+
+  // check coverage file
+  if(parse_json(coverage_file, message_handler, json))
+  {
+    messaget message(message_handler);
+    message.error() << coverage_file << " file is not a valid json file"
+                    << messaget::eom;
+    return true;
+  }
+
+  // make sure that we have an array of elements
+  if(!json.is_array())
+  {
+    messaget message(message_handler);
+    message.error() << "expecting an array in the " <<  coverage_file
+                    << " file, but got "
+                    << json << messaget::eom;
+    return true;
+  }
+
+  // traverse the given JSON file
+  for(const auto &goals_in_json : json.array)
+  {
+    // get the set of goals
+    if(goals_in_json["goals"].is_array())
+    {
+      // store the source location for each existing goal
+      for(const auto &each_goal : goals_in_json["goals"].array)
+      {
+        // get and set the bytecode_index
+        irep_idt bytecodeIndex=
+          each_goal["sourceLocation"]["bytecode_index"].value;
```

Comment:
Yes, it should be, but this induces a dependency on all the JSON changes that are currently in #1271. I'll mark this PR as do-not-merge until the JSON changes have been included.